### DISCUSSION TITLE
How do I configure a global mouse bind for scroll + keyboard modifier?

### DISCUSSION THREAD

**Discussion by MrDOS** [labwc/labwc#1106] [Q&A] [answered]
In Windows, I like to use [Volumouse](https://www.nirsoft.net/utils/volumouse.html) so that I can globally use Alt+mouse scroll up/down to quickly change the volume of my default audio output device. I can't remember exactly what equivalent I came up with the last time I used a Linux desktop, but I think it involved [pnmixer](https://github.com/nicklan/pnmixer); whatever it was, because it was running under X11, it was able to listen for mouse and keyboard events directly, always.

I'd like to mimic this feature in our brave new Wayland world. What's my best option for running a command in response to a mouse scroll event with a particular keyboard modifier in any context?

**Comment by MrDOS** [comment, marked as answer]
I don't see this explicitly documented anywhere, but the `direction` attribute of `<mousebind />` elements [is parsed for modifiers](https://github.com/labwc/labwc/blob/48e0b3f6a6bc7a60d8414f5f366a9d7bc3f1c42e/src/config/mousebind.c#L49-L58). Then it simply becomes a question of configuring two duplicate sets of mouse binds: one for when the mouse is over a window (“Frame” context), and one for when it isn't (“Root” context). It would be nice if there were a “Global” context, but I suspect it would only ever be useful for goofy stuff like this, so I'm not too fussed. All said and done, I've ended up with:

```xml
<labwc_config>
    <!-- 8< -->
    <mouse>
        <!-- 8< -->
        <context name="Frame">
            <mousebind direction="A-Up" action="Scroll">
                <action name="Execute" command="pactl set-sink-volume @DEFAULT_SINK@ +1%" />
            </mousebind>
            <mousebind direction="A-Down" action="Scroll">
                <action name="Execute" command="pactl set-sink-volume @DEFAULT_SINK@ -1%" />
            </mousebind>
        </context>

        <context name="Root">
            <mousebind direction="A-Up" action="Scroll">
                <action name="Execute" command="pactl set-sink-volume @DEFAULT_SINK@ +1%" />
            </mousebind>
            <mousebind direction="A-Down" action="Scroll">
                <action name="Execute" command="pactl set-sink-volume @DEFAULT_SINK@ -1%" />
            </mousebind>
        </context>
    </mouse>
</labwc_config>
```

...and I'm quite happy with it.

**Comment by Consolatis** [comment]
Yep, I think you found the optimal solution.

We only [mention](https://labwc.github.io/labwc-config.5.html#entry_mouse_context_name=_mousebind_button=_direction=_action=_action) the modifiers for `button` but not for the `direction` argument, should likely be updated to mention both.

Edit:
Fixed by
- #1147
